lk2nd/smb1360/smb1360-qcom.c

```diff
@@ -35,9 +41,9 @@ static const struct smb1360_battery *smb1360_check_batt_profile(const void *fdt,
 
 	rid10 = connected_rid / 10;
 
-	if ((profile_rid_a - connected_rid) < rid10) {
+	if (abs(profile_rid_a - connected_rid) < rid10) {
```

Comment:
> actually this won't cover all cases (seems wrong to me)

Why? Which one is not covered?

We want abs() here because we want to know how much the "connected_rid" is off from the expected ("profile_rid"). 

e.g. with `profile_rid_a = 100` we want to match approx. `90 < connected_rid < 110`. Therefore:

```
rid10 = 100 / 10 = 10;
100 - 100 = 0 < 10; // works
100 - 92 = 8 < 10; // works
100 - 108 = -8 < 10; // overflow with unsigned, works with your (int) suggestion
100 - 500 = -400 < 10; // overflow with unsigned, works incorrectly with your (int) suggestion
abs(100 - 108) = 8 < 10; // works
abs(100 - 500) = 400 < 10; // works
```

Reply:
oh you're right, I was a bit confused